packages/core/src/modules/cache/CachedStorageService.ts

```diff
@@ -0,0 +1,77 @@
+import { AgentContext } from '../../agent'
+import { BaseRecord } from '../../storage/BaseRecord'
+import { BaseRecordConstructor, Query, QueryOptions, StorageService } from '../../storage/StorageService'
+import { CacheModuleConfig } from './CacheModuleConfig'
+
+// biome-ignore lint/suspicious/noExplicitAny:
+export class CachedStorageService<T extends BaseRecord<any, any, any>> implements StorageService<T> {
+  public constructor(private storageService: StorageService<T>) {}
+
+  private cache(agentContext: AgentContext) {
+    return agentContext.resolve(CacheModuleConfig).cache
+  }
+
+  private getCacheKey(options: { type: string; id: string }) {
+    return `${options.type}:${options.id}`
+  }
+
+  public async save(agentContext: AgentContext, record: T): Promise<void> {
+    if (record.useCache) {
```

Comment:
I would call it `allowCache`, because it's only using cache if the use cached storage serivice is enabled